src/types/mod.rs

```diff
@@ -774,3 +902,178 @@ impl<T: Object> ValueInternal for JsFunction<T> {
+        let v: Handle<JsValue> = self.callee.do_call(cx, self.this, &self.args)?;
+        v.downcast_or_throw(cx)
+    }
+
+    /// Make the function call for side effect, discarding the result value. This method is
+    /// preferable to [`call()`](crate::types::FunctionCall::call) when the result value is
+    /// not needed, since it does not require specifying a result type.
+    pub fn exec<'b, C: Context<'b>>(&self, cx: &mut C) -> NeonResult<()> {
+        let _: Handle<JsValue> = self.callee.do_call(cx, self.this, &self.args)?;
+        Ok(())
+    }
+}
+
+impl<'a, T: Object> Call<'a, T> {
+    /// Set the value of `this` for the function call. Once a call has a `this` binding
+    /// specified, it is required to be a [`FunctionCall`](crate::types::FunctionCall).
+    pub fn this<V: Value>(self, this: Handle<'a, V>) -> FunctionCall<'a, T> {
+        FunctionCall {
+            callee: self.callee,
+            this: this.upcast(),
+            args: self.args,
+        }
+    }
+
+    /// Add an argument to the arguments list.
+    pub fn arg<V: Value>(&mut self, arg: Handle<'a, V>) -> &mut Self {
+        self.args.push(arg.upcast());
+        self
+    }
+
+    /// Add multiple arguments to the arguments list.
+    pub fn args<A: Arguments<'a>>(&mut self, args: A) -> &mut Self {
+        args.append(&mut self.args);
+        self
+    }
+
+    /// Call the function as a constructor (like a JavaScript `new` expression).
+    /// If the function returns without throwing, returns the resulting object.
+    pub fn new<'b, C: Context<'b>>(&self, cx: &mut C) -> JsResult<'b, T> {
+        self.callee.do_construct(cx, &self.args)
+    }
+
+    /// Make the function call. If the function returns without throwing, the result value
+    /// is downcast to the type `V`, throwing a `TypeError` if the downcast fails.
+    pub fn call<'b: 'a, C: Context<'b>, V: Value>(&self, cx: &mut C) -> JsResult<'b, V> {
+        let undefined: Handle<JsValue> = cx.undefined().upcast();
+        let v: Handle<JsValue> = self.callee.do_call(cx, undefined, &self.args)?;
+        v.downcast_or_throw(cx)
+    }
+
+    /// Make the function call for side effect, discarding the result value. This method is
+    /// preferable to [`call()`](crate::types::Call::call) when the result value is not
+    /// needed, since it does not require specifying a result type.
+    pub fn exec<'b: 'a, C: Context<'b>>(&self, cx: &mut C) -> NeonResult<()> {
+        let undefined: Handle<JsValue> = cx.undefined().upcast();
+        let _: Handle<JsValue> = self.callee.do_call(cx, undefined, &self.args)?;
+        Ok(())
+    }
+}
+
+/// The trait for specifying arguments in a [`Call`](crate::types::Call) or
+/// [`FunctionCall`](crate::types::FunctionCall). This trait is sealed and cannot
+/// be implemented by types outside of the Neon crate.
+pub trait Arguments<'a>: ArgumentsInternal {
+    #[doc(hidden)]
```

Comment:
Could this method be moved to `ArgumentsInternal` so that it doesn't need to be exposed to users (even doc hidden)?